### PR TITLE
mame & rom-tools: use system's expat

### PR DESCRIPTION
**Pull request by mbcoguno**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Recent macOS has expat 2.2, it's latest mahor and minor version. I think mame can use system's expat. I don't test the earlier expat (ships in Yosemite and earlier) can link or not. So Yosemite need Homebrew's expat right now.

~~Also update expat's keg_only reason.~~

### CODE REVIEW

**Comment by ilovezfs** [comment]
>I think mame can use system's expat. I don't test the earlier expat (ships in Yosemite and earlier) can link or not.

We should just assume it will be fine until proven otherwise.

File: Formula/expat.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@ class Expat < Formula
     sha256 "618683020e64ef1ca99d0c2f388262cf32117d93d7f047bf8251461d8af3f04e" => :el_capitan
   end
 
-  keg_only :provided_by_osx, "macOS includes Expat 1.5"
+  keg_only :provided_by_macos
```
Comment (ilovezfs) [inline Formula/expat.rb]: This should be in a separate pull request because it's not going to need new bottles.

Comment (mbcoguno) [reply Formula/expat.rb]: Homebrew/homebrew-core#20851

**Comment by ilovezfs** [comment]
Thanks again @mbcoguno!
